src/server/lib/upgradeConfig.js

```diff
@@ -14,11 +14,18 @@ module.exports = function (response) {
   var body = _.find(response.hits.hits, isUpgradeable);
   if (!body) return Promise.resolve();
 
+  // if the build number is still the template string (which it wil be in development)
+  // then we need to set it to the max interger. Otherwise we will set it to the build num
+  body._source.buildNum = (/^@@/.test(config.buildNum)) ?  Math.pow(2, 53) - 1 : parseInt(config.buildNum, 10);
```

Comment:
Is this perhaps more reliable than the Number.MAX_SAFE_INTEGER we use at https://github.com/elasticsearch/kibana/blob/master/src/kibana/index.js#L31 ? 

Maybe we should update it there as well?